install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -1072,6 +1072,7 @@ data:
 {{- if .Values.bgpControlPlane.enabled }}
   enable-bgp-control-plane: "true"
   bgp-secrets-namespace: {{ .Values.bgpControlPlane.secretsNamespace.name | quote }}
+  enable-bgp-v2-api: {{ .Values.bgpControlPlane.v2Enabled | quote }}
```

Comment:
Nit: How about `enable-bgp-control-plane-v2`? When we do v2-only, we set `enable-bgp-control-plane` to `false` right? Then it sounds a bit awkward to see that `bgp-control-plane` is disabled, but `v2-api` is enabled. WDYT?

Reply:
I was not planning to deprecate `enable-bgp-control-plane`. 

At the moment, I am thinking of this approach 
- We keep `enable-bgp-control-plane` to enable/disable BGP CP in its entirety. 
- `enable-bgp-v2-api` toggles if new API is used or not ( both in operator and agent )
- Once we give enough time, where both version will be supported ( PeeringPolicy and ClusterConfig ) - `enable-bgp-v2-api` will decide which configuration mode agent operates.
- When we decide to remove support for PeeringPolicy, we remove `enable-bgp-v2-api` flag ( or rather deprecate it  ). And by default ClusterConfig will be used .


Making completely independent `enable-bgp-control-plane-v2` would also mean duplicating some of the secret related flags - I wanted to avoid it. 

cc @rastislavs  since Yutaro is on vacation now. Want to brainstorm on this idea.

Reply:
@harsimran-pabla's approach makes sense to me - having `enable-bgp-v2-api` flag for just controlling which API is used, and eventually removing it after deprecating the old APIs.